sound/soc/sof/intel/hda-dai.c

```diff
@@ -400,6 +400,50 @@ static const struct snd_soc_dai_ops hda_link_dai_ops = {
 	.prepare = hda_link_pcm_prepare,
 };
 
+static struct sof_ipc_dai_config* ssp_dai_config_control(struct snd_sof_dev *sdev,
+							 struct snd_sof_dai *dai)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_control *scontrol;
+	struct sof_ipc_ctrl_data *cdata;
+	int number_configs = dai->number_configs;
+	int current_config = dai->current_config;
+	int index;
+
+	if (current_config > number_configs - 1) {
+		dev_err(sdev->dev, "error: DAI config index out of range\n");
+		return NULL;
+	}
+
+	config = &dai->dai_config[current_config];
+
+	/* return the only possible config if only 1 config present */
+	if (number_configs < 2)
```

Comment:
why not be explicit and say num_configs == 1

Reply:
ok, yes.